integration-tests/suites/base.go

```diff
@@ -119,20 +124,30 @@ func (s *IntegrationTestSuiteBase) StopCollector() {
 // one if it is nil. This function can be used to get the object before
 // the container is launched, so that Collector settings can be adjusted
 // by individual test suites
-func (s *IntegrationTestSuiteBase) Collector() *common.CollectorManager {
+func (s *IntegrationTestSuiteBase) Collector() (collector_manager.CollectorManager, error) {
 	if s.collector == nil {
-		s.collector = common.NewCollectorManager(s.Executor(), s.T().Name())
+		executor, err := s.Executor()
+		if err != nil {
+			return nil, err
+		}
+
+		s.collector = collector_manager.New(executor, s.T().Name())
 	}
-	return s.collector
+	return s.collector, nil
 }
 
 // Executor returns the current executor object, or initializes a new one
 // if it is nil.
-func (s *IntegrationTestSuiteBase) Executor() common.Executor {
+func (s *IntegrationTestSuiteBase) Executor() (executor.Executor, error) {
 	if s.executor == nil {
-		s.executor = common.NewExecutor()
+		exec, err := executor.New()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I think we can handle the error internally to keep this function as a cleanly-callable "getter" 

```suggestion
		s.Require().NoError(err)
```